Make error messages more consistent (step 1)

kocsismate opened the pull request:
Currently, error messages aren't consistent at all across all the extensions. I'd like to improve the situation by collecting some general principles that can be followed when writing/rewriting error messages.

For a start, find below some ideas that I could come up with. Any help or addition is welcome. :)

#### Capitalize all the error messages
If the message starts with a function/parameter/ini setting name (or something that is not suitable for capitalization), then either use `'` around them, or circumvent the issue (e.g. `enclosure must be a character` -> `The 'enclosure' parameter must be a single character` or `Argument 1 must be a single character`)

#### Use words consistently
E.g.: "cannot" or "can not", "filename" or "file name", "parameter" or "argument", "null" or "NUL".

#### Structure sentences consistently
E.g. Currently, we structure type error messages a lot of different ways:
- `Expects string, null, long or boolelan value for PostgreSQL '%s' (%s)` (sic!)
- `Expected array value for this option`
- `Array expected as options when using translate or scale`
- etc.

Instead of structuring messages all the possible ways, we should stick with similar structures.
For improving type error messages specifically, see [suggestion #5](#phrase-internal-errors-consistent-to-user-land-errors).

#### Use user provided values more consistently
Some example formats I found: `"Unable to fork [%s]"`, `"Type '%s' not supported"`, `Error opening file %s`, `"Invalid 'encoding' option - '%s'"` etc. Additionally, there are many cases when we don't even mention what the invalid user-provided value was: `Invalid OID specified`. I think we should try to display user-provided values when possible by using a common format.

#### Phrase internal errors consistent to user-land errors
In my opinion, error messages coming from internal functions should be similar to what is also reported for user-land functions/methods. E.g. instead of `Object or string expected`, `Argument 1 passed to foo() must be of the type object|string, array given` could be written. (Of course, in this case, the best solution would be to change the ZPP checks in question to use union types).

Another example: instead of `Requires 1 or 2 arguments`, we could write something similar to the user-land error message: `Too few arguments to function foo(), 0 passed and at least 1 expected`

This PR only contains capitalization changes so far - without any test fixes.

nikic commented on ext/calendar/cal_unix.c:
```diff
@@ -59,7 +59,7 @@ PHP_FUNCTION(jdtounix)
 
 	if (uday < 0 || uday > 24755) {
-		zend_value_error("jday must be within the Unix epoch");
+		zend_value_error("Jday must be within the Unix epoch");
```
This looks weird.

nikic commented on ext/date/php_date.c:
```diff
@@ -1101,7 +1101,7 @@ PHP_FUNCTION(idate)
 
 	if (ZSTR_LEN(format) != 1) {
-		php_error_docref(NULL, E_WARNING, "idate format is one char");
+		php_error_docref(NULL, E_WARNING, "Idate format is one char");
```
Weird

nikic commented on ext/ftp/ftp.c:
```diff
@@ -139,7 +139,7 @@ ftp_open(const char *host, short port, zend_long timeout_sec)
 	memset(&ftp->localaddr, 0, size);
 	if (getsockname(ftp->fd, (struct sockaddr*) &ftp->localaddr, &size) != 0) {
-		php_error_docref(NULL, E_WARNING, "getsockname failed: %s (%d)", strerror(errno), errno);
+		php_error_docref(NULL, E_WARNING, "Getsockname failed: %s (%d)", strerror(errno), errno);
```
Weird

nikic commented on ext/mbstring/mbstring.c:
```diff
@@ -3615,7 +3615,7 @@ php_mb_numericentity_exec(INTERNAL_FUNCTION_PARAMETERS, int type)
 	i = zend_hash_num_elements(target_hash);
 	if (i % 4 != 0) {
-		zend_value_error("count($convmap) must be a multiple of 4");
+		zend_value_error("Count($convmap) must be a multiple of 4");
```
Very weird

nikic commented on ext/mysqlnd/mysqlnd_plugin.c:
```diff
@@ -170,7 +170,7 @@ PHPAPI void mysqlnd_plugin_apply_with_argument(apply_func_arg_t apply_func, void
 		result = apply_func(val, argument);
 		if (result & ZEND_HASH_APPLY_REMOVE) {
-			php_error_docref(NULL, E_WARNING, "mysqlnd_plugin_apply_with_argument must not remove table entries");
+			php_error_docref(NULL, E_WARNING, "Mysqlnd_plugin_apply_with_argument must not remove table entries");
```
Weird

nikic commented on ext/spl/spl_directory.c:
```diff
@@ -2913,7 +2913,7 @@ SPL_METHOD(SplFileObject, fread)
 
 	if (length <= 0) {
-		php_error_docref(NULL, E_WARNING, "Length parameter must be greater than 0");
+		php_error_docref(NULL, E_WARNING, "length parameter must be greater than 0");
```
Wrong way around?

kocsismate replied:
My best guess why I did is that `length` is a parameter name - which should be lowercase. I'd like to improve this wording in the future though.

nikic commented on ext/standard/var.c:
```diff
@@ -1240,11 +1240,11 @@ PHP_FUNCTION(unserialize)
 		if (max_depth) {
 			if (Z_TYPE_P(max_depth) != IS_LONG) {
-				zend_type_error("max_depth should be int");
+				zend_type_error("Max_depth should be int");
```
Very weird (does not match case-sensitive spelling of the option name)

nikic commented on ext/sysvmsg/sysvmsg.c:
```diff
@@ -437,7 +437,7 @@ PHP_FUNCTION(msg_send)
 
 	if (result == -1) {
-		php_error_docref(NULL, E_WARNING, "msgsnd failed: %s", strerror(errno));
+		php_error_docref(NULL, E_WARNING, "Msgsnd failed: %s", strerror(errno));
```
Weird

nikic commented:
Not sure if I mentioned it to you before, but we have `scripts/dev/bless_tests.php` to mostly automate these kind of test updates.

kocsismate commented:
> Not sure if I mentioned it to you before, but we have scripts/dev/bless_tests.php to mostly automate these kind of test updates.

Ah, not yet! Thanks!

To answer you all the review comments at once: first, I started with mass-capitalization, and during doing so, I noticed some problematic patterns for which I tried to come up with a solution. The first suggestion in the description is in line with your comments, but I'd like to address them after we have decided about the possible directions (e.g. in this case: to stick with lower-case, or try to circumvent the issue).

So @nikic, @cmb69, @Girgias, if you are also interested in this topic, I'd love to hear your opinion about the proposals, and if you have anything else to add to the list.

Girgias commented:
On the top of my head, the length error/warning messages should also be consistent, because we currently have mix usage of ``should/must`` and different ways of formulating the same thing ``must be non negative`` and ``must be greater or equal than 0`` for example. I'm probably missing some others which I'll add when I encounter them again. :)

kocsismate commented:
Thanks for the idea, @Girgias , I agree with you.

I'll double check possible problems with capitalization + fix the tests so that this PR will be able to be merged, and I'll try to address the other suggestions in separate - and smaller - PRs afterwards.

nikic commented:
AppVeyor failures look legit.

kocsismate commented:
Thanks for the heads up! To sum up the changes since the last review: I tried to remove all the problematic/questionable changes, and made the tests green.

kocsismate commented on Zend/tests/dynamic_call_002.phpt:
```diff
@@ -1,5 +1,5 @@
 --TEST--
-Testing dynamic call with invalid value for method name
+Testing dynamic call with Invalid value for method name
```
Oops.

kocsismate commented on ext/oci8/oci8_interface.c:
```diff
@@ -424,7 +424,7 @@ PHP_FUNCTION(oci_lob_read)
 
 	if (length <= 0) {
-		php_error_docref(NULL, E_WARNING, "Length parameter must be greater than 0");
+		php_error_docref(NULL, E_WARNING, "length parameter must be greater than 0");
```
Makes sense, I was a bit uncertain about in which direction to go.

kocsismate commented on ext/phar/phar_object.c:
```diff
@@ -4438,7 +4438,7 @@ PHP_METHOD(Phar, extractTo)
 						case 0:
 							zend_throw_exception_ex(phar_ce_PharException, 0,
-								"Phar Error: attempted to extract non-existent file or directory \"%s\" from phar \"%s\"",
+								"phar error: attempted to extract non-existent file or directory \"%s\" from phar \"%s\"",
```
It's because the vast majority of such errors say "phar error", and just some of them say "Phar error" or "Phar Error". I am all for getting rid of these prefixes, but I'd prefer revisiting the problem later (if you have no objection).

nikic replied:
Sure, that's fine.
